coderd/database/queries/users.sql

```diff
@@ -128,7 +128,7 @@ WHERE
 
 -- name: GetUsers :many
 SELECT
-	*
+	*, COUNT(*) OVER() AS count
```

Comment:
I think you would still need to duplicate the filter without the limit right?

Reply:
I don't think so based on what I'm reading about OVER() https://stackoverflow.com/questions/156114/best-way-to-get-result-count-before-limit-was-applied/8242764#8242764

Reply:
This would make the query a lot less performant, but it seems _relatively_ standard based on my brazen scanning online.

Let's just write a benchmark in our SQL package to actually see how slow/fast it is at scale.

Reply:
I think it's objectively better than what is in main either way, since currently we just run the entire query twice, one without the filter.

Reply:
I can't imagine that postgres would run the `COUNT(*)` more than once so I'm happy with this direction. Would be good to test though.

Reply:
Agreed. Does this _just work_:tm:?

Reply:
Will report back when I can get some good tests.

Reply:
It looks like this does "just work". Need to fix a bunch of tests and stuff that changed with the type gen but my tests I've committed in https://github.com/coder/coder/pull/5016/commits/ed5a1473190d1c1a2b6767cc11d7f9776ae41479 are passing and working as expected.